osquery/events/darwin/openbsm.cpp

```diff
@@ -113,68 +115,139 @@ Status OpenBSMEventPublisher::setUp() {
   audit_pipe_ = fopen("/dev/auditpipe", "r");
   if (audit_pipe_ == nullptr) {
     LOG(WARNING) << "The auditpipe couldn't be opened.";
-    return Status::failure("Could not open OpenBSM pipe");
+    return Status::failure("Could not open auditpipe");
   }
 
   return FLAGS_audit_allow_config ? configureAuditPipe() : Status::success();
 }
 
-void OpenBSMEventPublisher::configure() {}
+void OpenBSMEventPublisher::configure() {
+  std::set<size_t> event_ids;
+  for (const auto& sub : subscriptions_) {
+    auto sc = getSubscriptionContext(sub->context);
+    event_ids.insert(sc->event_id);
+  }
+
+  WriteLock lock(event_ids_mutex_);
+  event_ids_ = event_ids;
```

Comment:
event_ids_ = std::move(event_ids);

Reply:
I can make this change but heads up I'm only really worried about optimizing the critical dequeue/parse path. A good follow up on the optimization route is looking at the `process_events` implementation on macOS and improving the string concatenation. (if you have more time to dedicate here)